crates/namada/src/vm/wasm/run.rs

```diff
@@ -602,18 +604,217 @@ where
     }
 }
 
-/// Get the gas rules used to meter wasm operations
-fn get_gas_rules() -> wasm_instrument::gas_metering::ConstantCostRules {
-    // NOTE: costs set to 0 don't actually trigger the injection of a call to
-    // the gas host function (no useless instructions are injected)
-    let instruction_cost = 0;
-    let memory_grow_cost = WASM_MEMORY_PAGE_GAS;
-    let call_per_local_cost = 0;
-    wasm_instrument::gas_metering::ConstantCostRules::new(
-        instruction_cost,
-        memory_grow_cost,
-        call_per_local_cost,
-    )
+struct GasRules;
+
+impl wasm_instrument::gas_metering::Rules for GasRules {
+    fn instruction_cost(
+        &self,
+        instruction: &wasm_instrument::parity_wasm::elements::Instruction,
+    ) -> Option<u32> {
+        // NOTE: costs set to 0 don't actually trigger the injection of a call
+        // to the gas host function (no useless instructions are
+        // injected)
+        // NOTE: these costs are taken from the benchmarks crate. None of them
+        // should be zero
+        let gas = match instruction {
+            Unreachable => 129_358,
```

Comment:
this makes aborting txs quite expensive, right? in the current scheme, we abort erroneous tx execution with a `panic!()`, that is passed back to the host as the guest entering unreachable code, since we do not have stack unwinding enabled in wasm

Reply:
Well if you look at the other gas costs you'll see that a tx on average consumes some hundreds millions unit of gas (which we then reduce for the actual fee payment), so even a few hundred thousand is not that much in the end. Also, fees are paid upfront, so it shouldn't metter how much gas is consumed by unreachable since the tx will fail anyway